src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainParameters.cs

```diff
@@ -9,6 +9,76 @@ namespace Nethermind.Specs.ChainSpecStyle;
 
 public class ChainParameters
 {
+
+    private ulong? _dencunTransitionTimestamp;
+    private ulong? _cancunTransitionTimestamp;
+
+    public ulong? DencunTransitionTimestamp
+    {
+        get
+        {
+            if (Eip4844TransitionTimestamp == Eip4788TransitionTimestamp &&
+                Eip4788TransitionTimestamp == Eip1153TransitionTimestamp &&
+                Eip1153TransitionTimestamp == Eip5656TransitionTimestamp &&
+                Eip5656TransitionTimestamp == Eip6780TransitionTimestamp)
```

Comment:
You could potentially use more declarative style of `Expression` Api to map individual properties to Hardfork property, that might be useful outside of this class too?

Reply:
Sure! In this refactor commit I created a dictionary at lines 15-22 so we can access only using the key tag, also a public static method GetHardforkMapping in case we need to get any value from this dictionary from any other classes outside